client/ayon_core/tools/sceneinventory/model.py

```diff
@@ -387,20 +425,38 @@ def set_hierarchy_view(self, state):
             self._hierarchy_view = state
 
     def get_outdated_item_ids(self, ignore_hero=True):
+        """Get item IDs of all outdated containers.
+        This method recursively searches the model hierarchy to find actual
+        container items (not groups) that are outdated.
+        """
         outdated_item_ids = []
-        root_item = self.invisibleRootItem()
-        for row in range(root_item.rowCount()):
-            group_item = root_item.child(row)
-            if group_item.data(VERSION_IS_LATEST_ROLE):
-                continue
 
-            if ignore_hero and group_item.data(VERSION_IS_HERO_ROLE):
-                continue
+        def collect_outdated_from_item(parent_item):
+            """Recursively collect outdated container item IDs."""
+            for row in range(parent_item.rowCount()):
+                item = parent_item.child(row)
+
+                # Check if this is an actual container item
+                is_container = item.data(IS_CONTAINER_ITEM_ROLE)
+
+                if is_container:
+                    # This is a container - check if it's outdated
+                    is_latest = item.data(VERSION_IS_LATEST_ROLE)
+                    is_hero = item.data(VERSION_IS_HERO_ROLE)
 
-            for idx in range(group_item.rowCount()):
-                item = group_item.child(idx)
-                outdated_item_ids.append(item.data(ITEM_ID_ROLE))
-        return outdated_item_ids
+                    if not is_latest and not (ignore_hero and is_hero):
+                        item_id = item.data(ITEM_ID_ROLE)
+                        if item_id:
+                            outdated_item_ids.append(item_id)
+                else:
+                    # This is a group item - recurse into its children
+                    collect_outdated_from_item(item)
+
+        root_item = self.invisibleRootItem()
+        # Collect outdated container ids from the full hierarchy
+        collect_outdated_from_item(root_item)
+        # Filter out any None values (e.g. from non-container rows)
+        return [item_id for item_id in outdated_item_ids if item_id]
```

Comment:
Check for item id already happens to be appended to `outdated_item_ids`.
```suggestion
        return outdated_item_ids
```